src/cuga/backend/knowledge/storage/adapter.py

```
@@ -104,47 +158,562 @@ def _cosine_similarity(distance: float) -> float:
+        except Exception as exc:  # noqa: BLE001 — defensive
+            logger.warning(
+                "BM25 FTS search failed for collection=%s query=%r: %s",
+                self._collection,
+                query[:40],
+                exc,
+            )
+            return []
+
+        out: list[tuple[Document, float]] = []
+        for row in rows:
+            chunk_id, source, filename, page, section_path, chunk_text, raw_rank = row
```

Ruff (0.15.18)

[warning] Unpacked variable `chunk_id` is never used

Prefix it with an underscore or any other dummy variable pattern

(RUF059)